From 45b61189358db327fe51872e060cdc590c4e1eb5 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Fri, 14 May 2021 12:38:57 +0200
Subject: [PATCH] Don't build tests unless requested by BUILD_TESTING

Variable is provided by KDECMakeSettings and default enabled.

Cleanup cmake a bit, Qt 6.6.0 is minimum by now.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt                                     | 4 +++-
 appiumtests/CMakeLists.txt                         | 9 ---------
 libdiscover/CMakeLists.txt                         | 4 +++-
 libdiscover/backends/DummyBackend/CMakeLists.txt   | 4 +++-
 libdiscover/backends/FlatpakBackend/CMakeLists.txt | 4 +++-
 libdiscover/backends/KNSBackend/CMakeLists.txt     | 4 +++-
 6 files changed, 15 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 95c3bd7e7..4da1df6a1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,7 +63,9 @@ add_definitions(-DQT_NO_SIGNALS_SLOTS_KEYWORDS -DQT_NO_URL_CAST_FROM_STRING)
 
 configure_file(DiscoverVersion.h.in DiscoverVersion.h)
 
-add_subdirectory(appiumtests)
+if(BUILD_TESTING AND CMAKE_SYSTEM_NAME MATCHES "Linux")
+    add_subdirectory(appiumtests)
+endif()
 add_subdirectory(libdiscover)
 add_subdirectory(discover)
 add_subdirectory(exporter)
diff --git a/appiumtests/CMakeLists.txt b/appiumtests/CMakeLists.txt
index d0214ec5c..8e610d3a4 100644
--- a/appiumtests/CMakeLists.txt
+++ b/appiumtests/CMakeLists.txt
@@ -1,15 +1,6 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # SPDX-FileCopyrightText: 2022 Harald Sitter <sitter@kde.org>
 
-if(NOT Qt6_VERSION VERSION_GREATER_EQUAL "6.5.0")
-    # Before Qt 6.5 the AbstractButtons didn't trigger properly at all.
-    return()
-endif()
-
-if(NOT BUILD_TESTING OR NOT CMAKE_SYSTEM_NAME MATCHES "Linux")
-    return()
-endif()
-
 find_package(SeleniumWebDriverATSPI)
 set_package_properties(SeleniumWebDriverATSPI PROPERTIES
     DESCRIPTION "Server component for selenium tests using Linux accessibility infrastructure"
diff --git a/libdiscover/CMakeLists.txt b/libdiscover/CMakeLists.txt
index 6e23a2f10..ff7038878 100644
--- a/libdiscover/CMakeLists.txt
+++ b/libdiscover/CMakeLists.txt
@@ -4,7 +4,9 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config-paths.h.cmake ${CMAKE_CURRENT_
 
 add_subdirectory(backends)
 add_subdirectory(notifiers)
-add_subdirectory(tests)
+if(BUILD_TESTING)
+    add_subdirectory(tests)
+endif()
 
 if (TARGET AppStreamQt)
     add_library(DiscoverAppStreamIntegration STATIC
diff --git a/libdiscover/backends/DummyBackend/CMakeLists.txt b/libdiscover/backends/DummyBackend/CMakeLists.txt
index 1c4fbea16..c8f2efeea 100644
--- a/libdiscover/backends/DummyBackend/CMakeLists.txt
+++ b/libdiscover/backends/DummyBackend/CMakeLists.txt
@@ -1,4 +1,6 @@
-add_subdirectory(tests)
+if(BUILD_TESTING)
+    add_subdirectory(tests)
+endif()
 
 set(dummy-backend_SRCS
     DummyResource.cpp
diff --git a/libdiscover/backends/FlatpakBackend/CMakeLists.txt b/libdiscover/backends/FlatpakBackend/CMakeLists.txt
index 9190b49dc..d20a66fef 100644
--- a/libdiscover/backends/FlatpakBackend/CMakeLists.txt
+++ b/libdiscover/backends/FlatpakBackend/CMakeLists.txt
@@ -17,7 +17,9 @@ target_link_libraries(libdiscover-backend-flatpak-logging-category
         Qt::Core
 )
 
-add_subdirectory(tests)
+if(BUILD_TESTING)
+    add_subdirectory(tests)
+endif()
 
 set(flatpak-backend_SRCS
     FlatpakResource.cpp
diff --git a/libdiscover/backends/KNSBackend/CMakeLists.txt b/libdiscover/backends/KNSBackend/CMakeLists.txt
index fe39611f5..4d0e59bf1 100644
--- a/libdiscover/backends/KNSBackend/CMakeLists.txt
+++ b/libdiscover/backends/KNSBackend/CMakeLists.txt
@@ -1,4 +1,6 @@
-add_subdirectory(tests)
+if(BUILD_TESTING)
+    add_subdirectory(tests)
+endif()
 
 add_library(kns-backend MODULE
     KNSBackend.cpp
-- 
2.44.0

